stdlib/2and3/_typeshed/__init__.pyi

```diff
@@ -52,9 +55,15 @@ OpenBinaryModeReading = Literal[
 ]
 OpenBinaryMode = Union[OpenBinaryModeUpdating, OpenBinaryModeReading, OpenBinaryModeWriting]
 
-@type_check_only
 class HasFileno(Protocol):
     def fileno(self) -> int: ...
 
 FileDescriptor = int
 FileDescriptorLike = Union[int, HasFileno]
+
+class SupportsRead(Protocol[_T_co]):
+    def read(self, __length: int = ...) -> _T_co: ...
+class SupportsReadline(Protocol[_T_co]):
+    def readline(self, __length: int = ...) -> _T_co: ...
```

Comment:
In practice, most implementations either supply the length arguments or they don't. I don't think it makes sense to split those protocols, though, and implementing reads that support both forms is fairly trivial.